### ISSUE TITLE
Is the torch.cuda.synchronize(self.device) necessary?

### ISSUE THREAD

**Issue by ginobilinie** [HuguesTHOMAS/KPConv-PyTorch#22]
Thanks for the great work. After reading the code in trainer.py (utils/trainer.py), I'd like to ask is torch.cuda.synchronize(self.device) necessary? Since it seems slower the training.

**Comment by HuguesTHOMAS** [comment]
Hi @ginobilinie,

I am not sure it is really necessary. I placed it there in my early experiments because I measure time spent in each part of the training loop and I wanted those measurements to be faithful. However, I am new to PyTorch and maybe this is useless. At the time, it did not slowed the training so I figured I could let it there. I havent tested it since so maybe it slows the network now.

Can I ask how much time you gain by removing torch.cuda.synchronize in your experiment?
